xprocess.py

```diff
@@ -47,17 +47,20 @@ def terminate(self, **kwargs):
         if not self.pid or not self.isrunning():
             return 0
         timeout = 20
```

Comment:
While we are updating this method, how about also turning this into a keyword-only parameter instead?

Reply:
Yes, being able to adjust how long to wait on process termination based on specificities of context (e.g what kind of external process your are running) makes sense for the user.